### DISCUSSION TITLE
[API] Refactoring for @file:ImportDataSchema: add some simple aliases for popular cases like CSV or JSON

### DISCUSSION THREAD

**Discussion by zaleslaw** [Kotlin/dataframe#263] [Ideas]
Now, the annotation @file:ImportDataSchema contains a lot of parameters that gives the user ability to cover all the possible scenarios. But probably it was started as a smaller feature but continued to grow and cover more scenarious (depends on the number of the supported data sources).

I suggest to refactor it in 2 steps: create two child annotations like @file:ImportDataSchemaFromCSV and @file:ImportDataSchemaFromJSON and work with them a little bit. If it's ok, we could make the basic annotation deprecated and remove after getting feedback from our users.

```
@file:ImportDataSchema(
    name = "Passenger",
    path = "src/main/resources/titanic.csv",
    csvOptions = CsvOptions(';')
)
```

What do you think @belovrv @Jolanrensen  @koperagen @ermolenkodev ?

**Comment by Jolanrensen** [comment]
Sounds like a good idea, especially since the autocomplete will show the user the types of data that can be accepted

**Comment by koperagen** [comment]
Honestly, I just followed the approach that we had with one general "read" function. It also was a way to automatically support newly added formats: with a separate annotation for each format, you'll have to explicitly support it in KSP plugin, right? It's going to be mostly repetitive code, but it will be something to keep in mind.
But! Recently I realized that we can write all kinds of fancy tests, even that check if someone forgot to support a format in a plugin. With this test, I can rest assured that nothing is going to be missed someday :)
